django/db/models/fields/__init__.py

```diff
@@ -857,10 +863,42 @@ def _get_default(self):
                 return self.default
             return lambda: self.default
 
+        if self.has_db_default():
+            return lambda: DB_DEFAULT
+
         if not self.empty_strings_allowed or self.null and not connection.features.interprets_empty_strings_as_nulls:
             return return_None
         return str  # return empty string
 
+    def has_db_default(self):
+        """Return a boolean of whether this field has a database default value."""
+        return self.db_default is not NOT_PROVIDED
+
+    def db_default_sql(self, model, schema_editor):
+        from django.db.models.expressions import Expression, Value
+        from django.db.models.functions import DefaultNow
+        from django.db.models.sql.query import Query
+
+        connection = schema_editor.connection
+        if connection.features.supports_functions_in_defaults and isinstance(self.db_default, Expression):
+            db_default = self.db_default
+            default_sql = '(%s)'
+        elif isinstance(self.db_default, DefaultNow):
+            db_default = self.db_default
+            default_sql = '%s'
+        else:
+            db_default = Value(self.db_default)
+            default_sql = '%s'
+
+        query = Query(model=model)
+        db_default = db_default.resolve_expression(query)
```

Comment:
This should not be necessary as not reference to non-litteral values should be allowed in here.